src/commonTest/kotlin/fr/acinq/lightning/channel/states/SpliceTestsCommon.kt

```diff
@@ -1169,11 +1380,80 @@ class SpliceTestsCommon : LightningTestSuite() {
+
+            val (alice1, bobQuiescent, spliceInit) = reachQuiescent(cmd, alice, bob)
+            val (bob1, actionsBob1) = bobQuiescent.process(ChannelCommand.MessageReceived(spliceInit))
+            val spliceAck = actionsBob1.findOutgoingMessage<SpliceAck>()
+            assertEquals(spliceAck.fundingContribution, 0.sat)
+            val (alice2, actionsAlice2) = alice1.process(ChannelCommand.MessageReceived(spliceAck))
+            val (alice4, bob3, actionsAlice4) = if (spliceIn_opt != null) {
+                val (bob2, actionsBob2) = bob1.process(ChannelCommand.MessageReceived(actionsAlice2.findOutgoingMessage<TxAddInput>()))
+                val (alice3, actionsAlice3) = alice2.process(ChannelCommand.MessageReceived(actionsBob2.findOutgoingMessage<TxComplete>()))
+                Triple(alice3, bob2, actionsAlice3)
+            } else {
+                Triple(alice2, bob1, actionsAlice2)
+            }
+
+            // alice adds shared input
+            val (bob4, actionsBob4) = bob3.process(ChannelCommand.MessageReceived(actionsAlice4.findOutgoingMessage<TxAddInput>()))
+            val (alice5, actionsAlice5) = alice4.process(ChannelCommand.MessageReceived(actionsBob4.findOutgoingMessage<TxComplete>()))
+
+            // alice adds shared output
+            val (bob5, actionsBob5) = bob4.process(ChannelCommand.MessageReceived(actionsAlice5.findOutgoingMessage<TxAddOutput>()))
+            val (alice6, actionsAlice6) = alice5.process(ChannelCommand.MessageReceived(actionsBob5.findOutgoingMessage<TxComplete>()))
+
+            val (alice8, bob6, actionsAlice8) = if (spliceOut_opt != null) {
+                val (bob6, actionsBob6) = bob5.process(ChannelCommand.MessageReceived(actionsAlice6.findOutgoingMessage<TxAddOutput>()))
+                val (alice7, actionsAlice7) = alice6.process(ChannelCommand.MessageReceived(actionsBob6.findOutgoingMessage<TxComplete>()))
+                Triple(alice7, bob6, actionsAlice7)
+            } else {
+                Triple(alice6, bob5, actionsAlice6)
+            }
+
+            val commitSigAlice = actionsAlice8.findOutgoingMessage<CommitSig>()
+            actionsAlice8.has<ChannelAction.Storage.StoreState>()
+            val (bob8, actionsBob8) = bob6.process(ChannelCommand.MessageReceived(actionsAlice8.findOutgoingMessage<TxComplete>()))
+            assertIs<LNChannel<Normal>>(bob8)
+            val commitSigBob = actionsBob8.findOutgoingMessage<CommitSig>()
+            actionsBob8.has<ChannelAction.Storage.StoreState>()
+
+            checkCommandResponse(cmd.replyTo, parentCommitment, spliceInit)
+            assertIs<LNChannel<Normal>>(alice8)
+            return UnsignedSpliceFixture(alice8, commitSigAlice, bob8, commitSigBob)
+        }
+
+        private fun initiateSplice(alice: LNChannel<Normal>, bob: LNChannel<Normal>, inAmounts: List<Satoshi> = emptyList(), pushAmount: MilliSatoshi = 0.msat, outAmount: Satoshi? = null): Pair<LNChannel<Normal>, LNChannel<Normal>> {
```

Comment:
It's more consistent with other functions to introduce one that combines splice-in and splice-out explicitly (not optionally like you're doing, which makes it hard to verify what exactly we're testing). Also, having a `pushAmount` here is overkill and makes it harder to check that the test does what it's supposed to do, we're explicitly testing the push mechanism in dedicated tests.